### PR TITLE
treewide (darwin): fix or enable darwin build for many packages (4)

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
See #114976 for details.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115756` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>analog</li>
    <li>coan</li>
    <li>lambda-mod-zsh-theme</li>
    <li>makefile2graph</li>
    <li>midicsv</li>
    <li>mnemonicode</li>
    <li>p910nd</li>
    <li>perl530Packages.AlienTidyp</li>
    <li>perl530Packages.HTMLTidy</li>
    <li>perl532Packages.AlienTidyp</li>
    <li>perl532Packages.HTMLTidy</li>
    <li>postsrsd</li>
    <li>python38Packages.tinyobjloader-py</li>
    <li>python39Packages.tinyobjloader-py</li>
    <li>r2mod_cli</li>
    <li>restya-board</li>
    <li>rocm-cmake</li>
    <li>routinator</li>
    <li>swarm</li>
    <li>tidyp</li>
    <li>tinyobjloader</li>
    <li>tllist</li>
    <li>udpt</li>
    <li>utf8cpp</li>
    <li>vcg</li>
  </ul>
</details>

**Comment by siraben** [comment]
Wowza!

File: pkgs/servers/routinator/default.nix [hunk 1]
```diff
@@ -11,13 +11,13 @@ rustPlatform.buildRustPackage rec {
...
 
+  buildInputs = lib.optionals stdenv.isDarwin [ Security ];
   cargoSha256 = "sha256-lhSSyJxxHc0t43xoDMtr/lSVL0xZl6poPYiyYXNvKKQ=";
 
   meta = with lib; {
     description = "An RPKI Validator written in Rust";
     homepage = "https://github.com/NLnetLabs/routinator";
     license = licenses.bsd3;
     maintainers = with maintainers; [ _0x4A6F ];
-    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/servers/routinator/default.nix:21]: I think we should set this to platforms.all instead.

File: pkgs/servers/udpt/default.nix [hunk 2]
```diff
@@ -21,7 +21,6 @@ rustPlatform.buildRustPackage rec {
     description = "A lightweight UDP torrent tracker";
     homepage = "https://naim94a.github.io/udpt";
     license = lib.licenses.gpl3;
-    platforms = lib.platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/servers/udpt/default.nix:24]: I think we should set this to platforms.all instead.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115756` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>analog</li>
<li>midicsv</li>
<li>p910nd</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
analog:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/admin/analog/default.nix:33:5:

```
   |
33 |     license = lib.licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115756` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>analog</li>
<li>coan</li>
<li>lambda-mod-zsh-theme</li>
<li>makefile2graph</li>
<li>midicsv</li>
<li>mnemonicode</li>
<li>p910nd</li>
<li>postsrsd</li>
<li>python38Packages.tinyobjloader-py</li>
<li>python39Packages.tinyobjloader-py</li>
<li>r2mod_cli</li>
<li>restya-board</li>
<li>rocm-cmake</li>
<li>routinator</li>
<li>swarm</li>
<li>tidyp</li>
<li>tinyobjloader</li>
<li>tllist</li>
<li>udpt</li>
<li>utf8cpp</li>
<li>vcg</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
analog:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/admin/analog/default.nix:33:5:

```
   |
33 |     license = lib.licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

File: pkgs/servers/routinator/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/servers/routinator/default.nix:21]: Done.

File: pkgs/servers/udpt/default.nix (hunk 2 above)
Comment (siraben) [reply pkgs/servers/udpt/default.nix:24]: Done.

**Comment by risicle** [comment]
Needs another rebase, but all builds for me on macos 10.14 - nice work.

**Comment by siraben** [comment]
Rebased.

**Comment by siraben** [comment]
@SuperSandro2000 ping

**Pull request merged**